### PR TITLE
Generate access token 

### PR DESCRIPTION
**Pull request by punitcse**
### What

We generate the access token from the github app so we dont need https://github.com/simplybusiness/approved-github-actions/pull/7

### Why

There are some security concern if we should use an external github action.

### CODE REVIEW

File: lib/config.rb [hunk 1]
```diff
@@ -2,15 +2,36 @@
...
+    bearer_client = Octokit::Client.new(bearer_token: bearer_token)
+    installation = bearer_client.find_repository_installation(payload['repository']['full_name'])
+    response = bearer_client.create_app_installation_access_token(installation[:id])
+    response[:token]
+  end
+
+  def bearer_token
+    payload = {
+      iat: Time.now.to_i,
+      exp: Time.now.to_i + (10 * 60),
```
Comment (sleepyfox) [inline lib/config.rb:30]: I'm guessing that this is 10 * 60 seconds?
Perhaps something like:
```suggestion
      exp: Time.now.to_i + (TEN_MINUTES),
```

File: lib/config.rb [hunk 2]
```diff
@@ -2,15 +2,36 @@
 
 require 'octokit'
 require 'json'
+require 'jwt'
```
Comment (sleepyfox) [inline lib/config.rb:5]:
```suggestion
require 'jwt'

TEN_MINUTES = 600 # seconds
```

**Review by sleepyfox — APPROVED** [review]

**Comment by DanHatesNumbers** [comment]
This seems to be following the Github recommended approach for performing actions an an installed app, my only concern is that there is no documentation about the private key that Dobby needs to perform actions, specifically how it should be setup for consumers to use. Otherwise happy with this approach.

**Review by DanHatesNumbers — APPROVED** [review]

**Comment by punitcse** [comment]
> This seems to be following the Github recommended approach for performing actions an an installed app, my only concern is that there is no documentation about the private key that Dobby needs to perform actions, specifically how it should be setup for consumers to use. Otherwise happy with this approach.

Thanks Dan I will add a separate PR for that.

**Pull request merged**